apps/web/components/webhook/WebhookDialogForm.tsx

```diff
@@ -29,20 +30,39 @@ export default function WebhookDialogForm(props: {
       subscriberUrl: "",
       active: true,
       payloadTemplate: null,
+      secret: null,
     } as Omit<TWebhook, "userId" | "createdAt" | "eventTypeId" | "appId">,
   } = props;
 
   const [useCustomPayloadTemplate, setUseCustomPayloadTemplate] = useState(!!defaultValues.payloadTemplate);
+  const [changeSecret, setChangeSecret] = useState(false);
+  const [newSecret, setNewSecret] = useState("");
+  const hasSecretKey = !!defaultValues.secret;
+  const currentSecret = defaultValues.secret;
 
   const form = useForm({
     defaultValues,
   });
+
+  const handleInput = (event: React.FormEvent<HTMLInputElement>) => {
+    setNewSecret(event.currentTarget.value);
+  };
+
+  useEffect(() => {
+    if (changeSecret) {
+      form.unregister("secret", { keepDefaultValue: false });
+    }
+  }, [changeSecret]);
+
   return (
     <Form
       data-testid="WebhookDialogForm"
       form={form}
       handleSubmit={async (event) => {
-        const e = { ...event, eventTypeId: props.eventTypeId };
+        const e = changeSecret
+          ? { ...event, eventTypeId: props.eventTypeId }
+          : { ...event, secret: currentSecret, eventTypeId: props.eventTypeId };
```

Comment:
Pass the old secret unless new one is set (or left blank  which suggests the user wishes to remove the secret key)